frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -28,49 +27,99 @@
 @PrepareForTest({GitMacheteErrorReportSubmitter.class})
 public class GitMacheteErrorReportSubmitterTest {
 
-  private String defaultErrorReport;
-  private String defaultErrorReportWithTemplate;
-
-  @Before
-  public void setUp() {
-    {
-      try {
-        defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
-        defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
-            StandardCharsets.UTF_8).strip();
-      } catch (IOException e) {
-        throw new RuntimeException(e);
-      }
-    }
+  // Used spy(...) here to still keep the original behaviour of the class and be able to stub getTemplateVariables method
+  // PowerMock API documentation Section 13 [https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/Mockito.html]
+  private static final GitMacheteErrorReportSubmitter gitMacheteErrorReportSubmitter = PowerMockito
+      .spy(new GitMacheteErrorReportSubmitter());
+  private static final LambdaLogger errorReportSubmitterLogger = PowerMockito.mock(LambdaLogger.class);
+  private static String defaultErrorReport;
+  private static String defaultErrorReportWithTemplate;
+  private static String errorReportWithStackTraceAndTitle;
+
+  @BeforeClass
+  @SneakyThrows
+  public static void setUp() {
+    defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
+    defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
+        StandardCharsets.UTF_8).strip();
+    errorReportWithStackTraceAndTitle = IOUtils.resourceToString("/error_report_uri_with_stacktrace_title.txt",
+        StandardCharsets.UTF_8).strip();
+
+    Whitebox.setInternalState(GitMacheteErrorReportSubmitter.class, errorReportSubmitterLogger);
```

Comment:
Wow nice... so it matches the value to the correct field (`LOG` in this case) automagically? 😯